core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -178,29 +182,38 @@ private void enqueueInvalidateDependentViews(TableToken viewToken, String invali
         }
     }
 
-    private SampleByIntervalIterator findSampleByIntervals(
+    private MatViewRefreshJob.RefreshIntervals findRefreshIntervals(
             @NotNull TableReader baseTableReader,
-            @NotNull MatViewDefinition viewDefinition,
+            @NotNull MatViewState viewState,
+            @NotNull MatViewTableMeta viewTableMeta,
             long lastRefreshTxn
     ) throws SqlException {
-        return findSampleByIntervals(baseTableReader, viewDefinition, lastRefreshTxn, Numbers.LONG_NULL, Numbers.LONG_NULL);
+        return findRefreshIntervals(baseTableReader, viewState, viewTableMeta, lastRefreshTxn, Numbers.LONG_NULL, Numbers.LONG_NULL);
     }
 
-    private SampleByIntervalIterator findSampleByIntervals(
+    private MatViewRefreshJob.RefreshIntervals findRefreshIntervals(
             @NotNull TableReader baseTableReader,
-            @NotNull MatViewDefinition viewDefinition,
+            @NotNull MatViewState viewState,
+            @NotNull MatViewTableMeta viewTableMeta,
             long lastRefreshTxn,
             long rangeFrom,
             long rangeTo
     ) throws SqlException {
+        refreshIntervals.clear();
```

Comment:
This method becomes nasty complicated. Can you please add comments with examples of what mat view refresh type (down to create mat view statements) go into what branch, and why?